source/common/upstream/upstream_impl.cc

```diff
@@ -59,9 +90,21 @@ Network::ClientConnectionPtr
 HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& cluster,
                            Network::Address::InstanceConstSharedPtr address,
                            const Network::ConnectionSocket::OptionsSharedPtr& options) {
+  Network::ConnectionSocket::OptionsSharedPtr cluster_options;
+  if (cluster.features() & ClusterInfo::Features::FREEBIND) {
+    cluster_options = std::make_shared<Network::ConnectionSocket::Options>();
+    if (options) {
+      *cluster_options = *options;
+    }
+    if (cluster.features() & ClusterInfo::Features::FREEBIND) {
```

Comment:
This line is redundant, it is already met in line 94.

Reply:
Thanks, that was some forest-and-trees code, will remove.